planning/obstacle_avoidance_planner/src/eb_path_optimizer.cpp

```diff
@@ -135,12 +135,28 @@ EBPathOptimizer::getOptimizedTrajectory(
     candidate_points.non_fixed_points.end());
 
   // interpolate points for logic purpose
-  const std::vector<geometry_msgs::msg::Point> interpolated_points =
+  std::vector<geometry_msgs::msg::Point> interpolated_points =
     interpolation_utils::getInterpolatedPoints(full_points, eb_param_.delta_arc_length_for_eb);
   if (interpolated_points.empty()) {
     return boost::none;
   }
 
+  // clip interpolated points with the length of path
+  if (traj_param_.enable_clipping_fixed_traj) {
+    if (path.points.size() < 2) {
+      return boost::none;
+    }
+    const auto interpolated_poses =
+      points_utils::convertToPosesWithYawEstimation(interpolated_points);
+    const auto interpolated_points_end_seg_idx = motion_utils::findNearestSegmentIndex(
+      interpolated_poses, path.points.back().pose, 3.0, 0.785);
```

Comment:
For **ego** nearest search, we can assume that the nearest point should be the first nearest one since the path length before the ego pose is short enough.
Other than that, we cannot assume that the nearest search should be the first nearest one.

Therefore, in this case, it's okay not to use `findfistNearestIndex`

Reply:
ok, understood thanks!